main.py

```diff
@@ -0,0 +1,140 @@
+
+    train, test = PreProcessor(dataset, lambda dataset: WaveformExtractor(dataset, column="audio", crop_length=2**18)).get_train_test_split(path=musiccaps_generator.get_processed_folder(), save_split_sets=False)
+
+    device = create_device()
+
+    batch_size = 8
+    transform = lambda x: x.unsqueeze(0)
+
+    train_dataset = TorchDataset(train, transform=transform)
+    train_dataloader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
+
+    autoencoder = create_model()
+
+    autoencoder = autoencoder.to(device)
+    optimizer = torch.optim.Adam(autoencoder.parameters(), lr=0.001)
+
+    checkpoint_path = "autoencoder_checkpoint.pth"
+    checkpoint_manager = ModelCheckpoint("autoencoder_checkpoint.pth")
+    start_epoch, start_batch_index, _ = checkpoint_manager.resume(autoencoder, optimizer)
+
+    num_epochs = 10
+    for epoch in tqdm(range(start_epoch, num_epochs), desc="Epochs"):
+        with tqdm(train_dataloader, unit="i", leave=False, desc="Batches") as tepoch:
+            for i, batch in enumerate(tepoch):    
+                if epoch == start_epoch and i < start_batch_index:
+                    continue
+
+                batch = batch.to(device)
+                optimizer.zero_grad()
+                loss = autoencoder(batch)
+                loss.backward()
+                optimizer.step()
+
+                if (i + 1) % 10 == 0:
+                    checkpoint_manager.save(autoencoder, optimizer, epoch, i, loss.item(), checkpoint_path)
+                    tepoch.set_description(f"Epoch {epoch} Batch {i+1}/{len(train_dataloader)} Loss: {loss.item():.4f} Saved")
+                else:
+                    # Update progress
+                    tepoch.set_description(f"Epoch {epoch} Batch {i+1}/{len(train_dataloader)} Loss: {loss.item():.4f}")
+    
+
+    # Save final model
+    final_model_path = "final_autoencoder_model.pth"
+    torch.save(autoencoder.state_dict(), final_model_path)
+    print(f"Training complete. The final model is saved to {final_model_path}")
+
+def save_audio(audio, sample_rate, filename):
+    write(filename, sample_rate, audio.cpu().numpy())
+
+def generate_music(model, device, num_samples=1, num_steps=1000):
+    audio = torch.randn(num_samples, 1, 2**18, device=device)
+
+    audio = audio.to(next(model.parameters()).device)
+
+    model.eval() 
+
+    with torch.no_grad():
+        latent = model.encode(audio)
+        generated_audio = model.decode(latent, num_steps=num_steps)
+
+    return generated_audio.cpu()
+
+if __name__ == "__main__":
+
+    final_model_path = "final_autoencoder_model.pth"
+
+    if not os.path.isfile(final_model_path):
+        train(final_model_path)
+
+    device = create_device()
+    model = create_model()
+    model.load_state_dict(torch.load(final_model_path, map_location=device))
+    model = model.to(device)
+    
+    print("Generating music...")
+    generated_audio = generate_music(model, device)
```

Comment:
Você disse que gerou ruído, mas também enviou ruído, então fica difícil experimentar sem uma música real, em vez disso considere usar uma música do dataset de test.
> Antigamente: 
```python
 audio = torch.randn(num_samples, 1, 2**18, device=device)
```
> Sugestão:

```suggestion
    audio = test[:1].unsqueeze(0)
    audio  = torch.from_numpy(audio )
    generated_audio = generate_music(model, device, audio)
```